misc/amia_paper_2024/stats.py

```diff
@@ -0,0 +1,60 @@
+"""AMIA paper 2024: Statistics
+
+Resources
+- GH issue: https://github.com/jhu-bids/TermHub/issues/631
+- Paper: https://livejohnshopkins-my.sharepoint.com/:w:/g/personal/sgold15_jh_edu/EXbuxwBsb0pPgFfgPYjR6qgBlJ1gmpqsMPYJ3vn6Y_HnYw?e=4%3ApEDbMe&fromShare=true&at=9&CID=d4663c40-1ebe-db59-8cf5-2bc091d45692"""
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List
+
+import pandas as pd
+
+THIS_DIR = Path(os.path.dirname(__file__))
+PROJECT_ROOT = THIS_DIR.parent.parent
+sys.path.insert(0, str(PROJECT_ROOT))
+from backend.routes.db import usage_query
+
+
+OUTDIR = THIS_DIR / 'output'
+CSV_PATH = OUTDIR / 'usage.csv'
+
+
+def setup():
+    """Setup any environmental prerequisites"""
+    if not OUTDIR.exists():
+        OUTDIR.mkdir()
+
+
+def get_data() -> pd.DataFrame:
+    """Get data to be analyzed"""
+    if not os.path.exists(CSV_PATH):
+        # JSON_FILE = '/Users/joeflack4/Desktop/vs-hub-usage-report.json'
+        # with open(JSON_FILE, 'r') as file:
+        #     d = json.load(file)
+        # Fetch
+        data: List[Dict] = usage_query()
+        df = pd.DataFrame(data)
+        # Save
+        t0 = datetime.now()
+        df.to_csv(CSV_PATH, index=False)
+        print(f'Wrote {len(df)} records to CSV in n seconds: {(datetime.now() - t0).seconds}')
+    else:
+        # Read
+        t0 = datetime.now()
+        df = pd.read_csv(CSV_PATH)
```

Comment:
## CSV
Reads CSV if it exists in local (git ignored) dir. Else, it fetches data and writes the CSV. FYI on my PC it takes 7 seconds to read the CSV. Also the CSV is a similar size to the JSON; almost 3GB.